components/eam/src/chemistry/aerosol/cldaero_mod.F90

```diff
@@ -79,49 +79,60 @@ end subroutine cldaero_deallocate
 ! utility function for cloud-borne aerosols
 !----------------------------------------------------------------------------------
 
+!==================================================================================
   function cldaero_uptakerate( xl, cldnum, cfact, cldfrc, tfld,  press ) result( uptkrate )
-    use mo_constants, only : pi
-
-    real(r8), intent(in) :: xl, cldnum, cfact, cldfrc, tfld,  press
-
-    real(r8) :: uptkrate
-
-    real(r8) :: &
-         rad_cd, radxnum_cd, num_cd, &
-         gasdiffus, gasspeed, knudsen, &
-         fuchs_sutugin, volx34pi_cd
-
 !-----------------------------------------------------------------------
 ! compute uptake of h2so4 and msa to cloud water
 !
 ! first-order uptake rate is
 ! 4*pi*(drop radius)*(drop number conc)
 ! *(gas diffusivity)*(fuchs sutugin correction)
+!-----------------------------------------------------------------------
+
+    use mo_constants, only : pi
 
-! num_cd = (drop number conc in 1/cm^3)
+    ! input arguments
+    real(r8), intent(in) :: xl          ! liquid water volume [cm^3/cm^3]
+    real(r8), intent(in) :: cldnum      ! droplet number concentration [#/kg]
+    real(r8), intent(in) :: cfact       ! total atms density [kg/L]
+    real(r8), intent(in) :: cldfrc      ! cloud fraction [fraction]
+    real(r8), intent(in) :: tfld        ! temperature [K]
+    real(r8), intent(in) :: press       ! pressure [Pa]
+    ! output arguments
+    real(r8) :: uptkrate        ! uptake rate [cm/cm/s]
+    ! local variables
+    real(r8) :: rad_cd          ! droplet radius [cm]
+    real(r8) :: radxnum_cd      ! radius * number conc [cm/cm^3]
+    real(r8) :: num_cd          ! droplet number conc [1/cm^3]
+    real(r8) :: gasdiffus       ! H2SO4 gas diffusivity [cm^2/s]
+    real(r8) :: gasspeed        ! H2SO4 gas mean molecular speed [cm/s]
+    real(r8) :: knudsen         ! knudsen number [unitless]
+    real(r8) :: fuchs_sutugin   ! another dimensionless number
+    real(r8) :: volx34pi_cd     ! droplet volume * 3/4*pi [cm^3/cm^3]
+
+    real(r8),parameter :: one_third = 0.3333333_r8
+
+! change drop number conc from #/kg to #/cm^3
         num_cd = 1.0e-3_r8*cldnum*cfact/cldfrc
         num_cd = max( num_cd, 0.0_r8 )
```

Comment:
We can make 1.0e-3_r8 a unit conversion parameter. It think it is widely used and might be eventually made as a common parameter for all subroutines.